exporters/otlp/otlptrace/internal/tracetransform/span.go

```diff
@@ -146,16 +147,28 @@ func links(links []tracesdk.Link) []*tracepb.Span_Link {
 		tid := otLink.SpanContext.TraceID()
 		sid := otLink.SpanContext.SpanID()
 
+		flags := buildSpanFlags(otLink.SpanContext)
+
 		sl = append(sl, &tracepb.Span_Link{
 			TraceId:                tid[:],
 			SpanId:                 sid[:],
 			Attributes:             KeyValues(otLink.Attributes),
 			DroppedAttributesCount: uint32(otLink.DroppedAttributeCount),
+			Flags:                  flags,
 		})
 	}
 	return sl
 }
 
+func buildSpanFlags(sc trace.SpanContext) uint32 {
+	flags := tracepb.SpanFlags_SPAN_FLAGS_CONTEXT_HAS_IS_REMOTE_MASK
+	if sc.IsRemote() {
+		flags = flags | tracepb.SpanFlags_SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK
```

Comment:
Wow, these `tracepb` const names are confusing! :laughing: 

Can you clarify here what `tracepb.SpanFlags_SPAN_FLAGS_CONTEXT_HAS_IS_REMOTE_MASK`? From the [docs](https://github.com/open-telemetry/opentelemetry-proto-go/blob/main/otlp/trace/v1/trace.pb.go):

> // Bits 8 and 9 are used to indicate that the parent span or link span is remote.
> // Bit 8 (`HAS_IS_REMOTE`) indicates whether the value is known.
> // Bit 9 (`IS_REMOTE`) indicates whether the span or link is remote.

it seems like we need to know if the remove value is known, but that may not always be the case. The default of a span context is non-remote, but that may just be the default being used and not a user saying they known the span is non-remote. Rigth?

Reply:
Yes, your assumption is correct.

The thing is, we don't have any way to know whether the span is non-remote because the user knows about it, or because it's actually non-remote.
I'm also not sure this is actually an issue, as the issue could arise only if someone is creating span contexts by themselves, and misses setting the parent's remote information.

In [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isremote), properly specifying `IsRemote` is also a MUST.